Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/FieldedMetadataTableRow.java

```diff
@@ -0,0 +1,448 @@
+     * if the value is ambiguous or cannot be casted.
+     *
+     * @param values
+     *            values obtained
+     * @return the only entry or null
+     */
+    private MetadataEntry oneSimpleValue(Collection<Metadata> values) {
+        switch (values.size()) {
+            case 0:
+                return null;
+            case 1: {
+                Metadata metadata = values.iterator().next();
+                if (metadata instanceof MetadataEntry) {
+                    return (MetadataEntry) metadata;
+                } else {
+                    throw new IllegalStateException("Got complex meta-data entry with key \"" + metadata.getKey()
+                            + "\" which isn't declared as substructured key in the rule set.");
+                }
+            }
+            default:
+                throw new IllegalStateException("Too many (" + values.size() + ") meta-data of type \""
+                        + values.iterator().next().getKey() + "\" in a single row. Must be 0 or 1 per row.");
+        }
+    }
+
+    void addAdditionallySelectedField(String additionallySelectedField)
+            throws InvalidMetadataValueException, NoSuchMetadataFieldException {
+        additionallySelectedFields.add(additionallySelectedField);
+        preserve();
+        createMetadataTable();
+    }
+
+    /**
+     * Returns the elements for the drop-down addable meta-data picklist.
+     *
+     * @return the addable meta-data elements
+     */
+    List<SelectItem> getAddableMetadata() {
+        Map<Metadata, String> metadataWithKeys = addLabels(metadata).parallelStream()
+                .collect(Collectors.toMap(Function.identity(), Metadata::getKey));
+        return metadataView.getAddableMetadata(metadataWithKeys, additionallySelectedFields).stream()
+                .map(addable -> new SelectItem(addable.getId(), addable.getLabel())).collect(Collectors.toList());
+    }
+
+    @Override
+    public String getInput() {
+        return "dataTable";
+    }
+
+    /**
+     * Returns the meta-data of a meta-data group, when used recursively.
+     *
+     * @return the meta-data of the meta-data group
+     * @throws InvalidMetadataValueException
+     *             if some value is invalid
+     */
+    @Override
+    Collection<Metadata> getMetadata() throws InvalidMetadataValueException {
+        assert structure == null;
+        MetadataGroup result = new MetadataGroup();
+        result.setKey(metadataView.getId());
+        result.setDomain(DOMAIN_TO_MDSEC.get(metadataView.getDomain().orElse(Domain.DESCRIPTION)));
+        try {
+            this.preserve();
+        } catch (NoSuchMetadataFieldException e) {
+            /*
+             * The domain attribute is not evaluated on members of meta-data
+             * groups, so writing to fields can never happen here and thus
+             * cannot cause the exception.
+             */
+            throw new IllegalStateException("never happening exception");
```

Comment:
Yes, the domains are separate areas in the XML. So, if the group of meta-data is in an element `<amdSec><techMD>`, then a single meta-data from it cannot be in the element <dmdSec>, since it has to be inside its parent XML element.